mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp

```diff
@@ -361,7 +361,7 @@ struct ClampClampOptimization : public OpRewritePattern<tosa::ClampOp> {
           op, op.getType(), clampOp.getInput(),
           rewriter.getI64IntegerAttr(minInt),
           rewriter.getI64IntegerAttr(maxInt), rewriter.getF32FloatAttr(minFp),
-          rewriter.getF32FloatAttr(maxFp));
+          rewriter.getF32FloatAttr(maxFp), rewriter.getStringAttr("IGNORE"));
```

Comment:
Why is `IGNORE` set here?

Reply:
I believe we need to be more careful with how we handle the NaN mode here. 

I'll need to spend some more time checking, but so far the following seems to be giving the correct results:
```
- op = PROPAGATE, clampOp = PROPAGATE -> resultingOp = PROPAGATE
- op = IGNORE, clampOp = IGNORE -> resultingOp = IGNORE
- op = PROPAGATE, clampOp = IGNORE -> don't perform optimization
- op = IGNORE, clampOp = PROPAGATE -> resultingOp = IGNORE
```